src/app/backend/handler/apihandler.go

```diff
@@ -173,13 +171,38 @@ func FormatResponseLog(resp *restful.Response, req *restful.Request) string {
 		req.Request.RemoteAddr, resp.StatusCode())
 }
 
-// CreateHTTPAPIHandler creates a new HTTP handler that handles all requests to the API of the backend.
-func CreateHTTPAPIHandler(client *clientK8s.Clientset, heapsterClient client.HeapsterClient,
-	clientConfig clientcmd.ClientConfig) (http.Handler, error) {
+func (apiHandler *APIHandler) injectApiClient(fn func(*clientK8s.Clientset, *restful.Request, *restful.Response)) restful.RouteFunction {
+	return func(request *restful.Request, response *restful.Response) {
+		cfg, err := apiHandler.clientConfig.ClientConfig()
```

Comment:
Are you allowed to modify the results of `ClientConfig()`? I don't see anything in the godoc of the interface. Maybe play it safe and make a shallow clone to ensure modifying the `BearerToken` wont impact other callers? Or add a test to ensure it doesn't.